presto-product-tests/src/main/java/io/prestosql/tests/cli/PrestoCliFileAuthTests.java

```diff
@@ -0,0 +1,190 @@
+        Thread.sleep(10); // pausing till refresh-period for reloading the file
+        launchPrestoCliWithServerArgument("userBCrypt1", "user123", "--execute", "select * from hive.default.nation;");
+        assertSuccess();
+        //delete the content in file and check for failure
+        deleteContentToFile("/docker/volumes/conf/presto/etc/file_auth_copy", "userBCrypt1:$2y$10$BqTb8hScP5DfcpmHo5PeyugxHz5Ky/qf3wrpD7SNm8sWuA3VlGqsa");
+        launchPrestoCliWithServerArgument("userBCrypt1", "user123", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    @Test(groups = {FILEAUTH, PROFILE_SPECIFIC_TESTS}, timeOut = TIMEOUT)
+    public void testNonExistentUserFailure()
+            throws IOException, InterruptedException
+    {
+        launchPrestoCliWithServerArgument("userDNE", "passwordXYZ", "--execute", "select * from hive.default.nation;");
+        assertAuthenticationError();
+    }
+
+    private void assertSuccess()
+    {
+        assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
+        skipAfterTestWithContext();
+    }
+
+    private void assertAuthenticationError()
+    {
+        assertTrue(trimLines(presto.readRemainingErrorLines()).stream().anyMatch(str -> str.contains("Authentication failed")));
+        skipAfterTestWithContext();
+    }
+
+    private void skipAfterTestWithContext()
+    {
+        if (presto != null) {
+            presto.close();
+            presto = null;
+        }
+    }
+
+    private void addContentToFile(String filename, String content)
+    {
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(filename, true)); //clears file every time
+            writer.newLine(); //Add new line
+            writer.write(content);
+            writer.close();
+        }
+        catch (Exception e) {
+            System.err.println(e);
+        }
+    }
+
+    private void deleteContentToFile(String filename, String content)
+    {
+        try {
+            List<String> stream = Files.lines(Paths.get(filename)).filter(s -> !s.contains(content)).collect(Collectors.toList());
+            Files.write(Paths.get(filename), stream);
+        }
+        catch (Exception e) {
+            System.err.println(e);
+        }
+    }
+
+    private void launchPrestoCliWithServerArgument(String username, String password, String... arguments)
+            throws IOException, InterruptedException
+    {
+        requireNonNull(truststorePath, "databases.presto.cli_truststore_path is null");
+        requireNonNull(truststorePassword, "databases.presto.cli_truststore_password is null");
+        requireNonNull(serverAddress, "databases.presto.cli_server_address is null");
+
+        ImmutableList.Builder<String> prestoClientOptions = ImmutableList.builder();
```

Comment:
We can simplify this by doing the build all at once
```java
List<String> prestoClientOptions = ImmutableList.builder()
        .add("--server", serverAddress)
        .add(...)
        .add(arguments)
        .build();

Reply:
Done

Reply:
`List<String> prestoClientOptions = ImmutableList.builder()
                .add("--server", serverAddress)
                .add("--truststore-path", truststorePath,
                "--truststore-password", truststorePassword,
                "--user", username,
                "--password")
                .add(arguments)
                .build()
                .stream()
                .map(obj -> obj.toString())
                .collect(Collectors.toList());`